merlin/spec/specification.py

```diff
@@ -301,6 +240,85 @@ def dict_to_yaml(obj, string, key_stack, newline=True):
             raise ValueError(f"Error parsing provenance spec:\n{e}")
         return result
 
+    def _dict_to_yaml(self, obj, string, key_stack, tab, newline=True):
+        """
+        The if-else ladder for sorting the yaml string prettification of dump().
+        """
+        if obj is None:
+            return ""
+
+        lvl = len(key_stack) - 1
+
+        if isinstance(obj, str):
+            return self._process_string(obj, lvl, tab)
+        elif isinstance(obj, bool):
+            return str(obj).lower()
+        elif not (isinstance(obj, list) or isinstance(obj, dict)):
+            return obj
+        else:
+            return self._process_dict_or_list(obj, string, key_stack, lvl, tab)
+
+    def _process_string(self, obj, lvl, tab):
+        """
+        Processes strings for _dict_to_yaml() in the dump() method.
+        """
+        split = obj.splitlines()
+        if len(split) > 1:
+            obj = "|\n" + tab * (lvl + 1) + ("\n" + tab * (lvl + 1)).join(split)
+        return obj
+
+    def _process_dict_or_list(self, obj, string, key_stack, lvl, tab):
```

Comment:
i find this function hard to follow. i know it was hard to follow before and maybe i'm just staring at it for the first tie

Reply:
@AlexanderWinterLLNL I'm in favor of decomposition, but both versions are hard to follow at a glance. So can you verify that the "pretty" provenence spec generated by your changes is the same as before your changes?

Reply:
@AlexanderWinterLLNL I just ran this test on my personal machine w/ feature_demo, and both version indeed produce  identical provenance specs. Good job